### ISSUE TITLE
Running winetricks creates thousands of folders in a mounted drive

### ISSUE THREAD

**Issue by txtsd** [Winetricks/winetricks#1688]
Alright so I have an external drive mounted at `/run/media/<user>/<UUID>`

When I run `WINEPREFIX=$prefixdir winetricks $verb` with and without `--unattended` or a `$verb`, it stalls and generates thousands of folders in my external drive.

Here's an example log of winetricks:
```
λ WINEPREFIX=/home/txtsd/1TB/Games/kingdoms-of-amalur-reckoning winetricks
Executing mkdir -p /home/txtsd/1TB/Games
------------------------------------------------------
warning: You are using a 64-bit WINEPREFIX. Note that many verbs only install 32-bit versions of packages. If you encounter problems, please retest in a clean 32-bit WINEPREFIX before reporting a bug.
------------------------------------------------------
Using winetricks 20201206 - sha256sum: e403ec830c40675225e1d6c1b78cc2da54376ddbeea0405b62b1b9576bba49d1 with wine-6.0.r0.g2414b1da ( TkG Staging Esync Fsync ) and WINEARCH=win64
winetricks GUI enabled, using zenity 3.32.0
Executing w_do_call vcrun2008
Executing mkdir -p /home/txtsd/1TB/Games
------------------------------------------------------
warning: You are using a 64-bit WINEPREFIX. Note that many verbs only install 32-bit versions of packages. If you encounter problems, please retest in a clean 32-bit WINEPREFIX before reporting a bug.
------------------------------------------------------
Executing load_vcrun2008 
Using native,builtin override for following DLLs: atl90 msvcm90 msvcp90 msvcr90 vcomp90
Executing rm -rf /home/txtsd/1TB/Games/kingdoms-of-amalur-reckoning/dosdevices/c:/windows/winsxs/manifests/amd64_microsoft.vc90.atl_1fc8b3b9a1e18e3b_9.0.30729.6161_none_deadbeef.manifest
Executing rm -rf /home/txtsd/1TB/Games/kingdoms-of-amalur-reckoning/dosdevices/c:/windows/winsxs/manifests/amd64_microsoft.vc90.crt_1fc8b3b9a1e18e3b_9.0.30729.6161_none_deadbeef.manifest
Executing rm -rf /home/txtsd/1TB/Games/kingdoms-of-amalur-reckoning/dosdevices/c:/windows/winsxs/manifests/x86_microsoft.vc90.atl_1fc8b3b9a1e18e3b_9.0.30729.6161_none_deadbeef.manifest
Executing rm -rf /home/txtsd/1TB/Games/kingdoms-of-amalur-reckoning/dosdevices/c:/windows/winsxs/manifests/x86_microsoft.vc90.crt_1fc8b3b9a1e18e3b_9.0.30729.6161_none_deadbeef.manifest
Executing rm -rf /home/txtsd/1TB/Games/kingdoms-of-amalur-reckoning/dosdevices/c:/windows/winsxs/manifests/x86_microsoft.vc90.mfc_1fc8b3b9a1e18e3b_9.0.30729.6161_none_deadbeef.manifest
Executing rm -rf /home/txtsd/1TB/Games/kingdoms-of-amalur-reckoning/dosdevices/c:/windows/winsxs/manifests/x86_microsoft.vc90.mfcloc_1fc8b3b9a1e18e3b_9.0.30729.6161_none_deadbeef.manifest
Executing rm -rf /home/txtsd/1TB/Games/kingdoms-of-amalur-reckoning/dosdevices/c:/windows/winsxs/manifests/x86_microsoft.vc90.openmp_1fc8b3b9a1e18e3b_9.0.30729.6161_none_deadbeef.manifest
Executing rm -rf /home/txtsd/1TB/Games/kingdoms-of-amalur-reckoning/dosdevices/c:/windows/winsxs/manifests/amd64_microsoft.vc90.atl_1fc8b3b9a1e18e3b_9.0.30729.6161_none_deadbeef.manifest
Executing rm -rf /home/txtsd/1TB/Games/kingdoms-of-amalur-reckoning/dosdevices/c:/windows/winsxs/manifests/amd64_microsoft.vc90.crt_1fc8b3b9a1e18e3b_9.0.30729.6161_none_deadbeef.manifest
Executing rm -rf /home/txtsd/1TB/Games/kingdoms-of-amalur-reckoning/dosdevices/c:/windows/winsxs/manifests/x86_microsoft.vc90.atl_1fc8b3b9a1e18e3b_9.0.30729.6161_none_deadbeef.manifest
Executing rm -rf /home/txtsd/1TB/Games/kingdoms-of-amalur-reckoning/dosdevices/c:/windows/winsxs/manifests/x86_microsoft.vc90.crt_1fc8b3b9a1e18e3b_9.0.30729.6161_none_deadbeef.manifest
Executing rm -rf /home/txtsd/1TB/Games/kingdoms-of-amalur-reckoning/dosdevices/c:/windows/winsxs/manifests/x86_microsoft.vc90.mfc_1fc8b3b9a1e18e3b_9.0.30729.6161_none_deadbeef.manifest
Executing rm -rf /home/txtsd/1TB/Games/kingdoms-of-amalur-reckoning/dosdevices/c:/windows/winsxs/manifests/x86_microsoft.vc90.mfcloc_1fc8b3b9a1e18e3b_9.0.30729.6161_none_deadbeef.manifest
Executing rm -rf /home/txtsd/1TB/Games/kingdoms-of-amalur-reckoning/dosdevices/c:/windows/winsxs/manifests/x86_microsoft.vc90.openmp_1fc8b3b9a1e18e3b_9.0.30729.6161_none_deadbeef.manifest
Executing rm -rf /home/txtsd/1TB/Games/kingdoms-of-amalur-reckoning/dosdevices/c:/windows/winsxs/manifests/amd64_microsoft.vc90.atl_1fc8b3b9a1e18e3b_9.0.30729.6161_none_deadbeef.manifest
Executing rm -rf /home/txtsd/1TB/Games/kingdoms-of-amalur-reckoning/dosdevices/c:/windows/winsxs/manifests/amd64_microsoft.vc90.crt_1fc8b3b9a1e18e3b_9.0.30729.6161_none_deadbeef.manifest
Executing rm -rf /home/txtsd/1TB/Games/kingdoms-of-amalur-reckoning/dosdevices/c:/windows/winsxs/manifests/x86_microsoft.vc90.atl_1fc8b3b9a1e18e3b_9.0.30729.6161_none_deadbeef.manifest
Executing rm -rf /home/txtsd/1TB/Games/kingdoms-of-amalur-reckoning/dosdevices/c:/windows/winsxs/manifests/x86_microsoft.vc90.crt_1fc8b3b9a1e18e3b_9.0.30729.6161_none_deadbeef.manifest
Executing rm -rf /home/txtsd/1TB/Games/kingdoms-of-amalur-reckoning/dosdevices/c:/windows/winsxs/manifests/x86_microsoft.vc90.mfc_1fc8b3b9a1e18e3b_9.0.30729.6161_none_deadbeef.manifest
Executing rm -rf /home/txtsd/1TB/Games/kingdoms-of-amalur-reckoning/dosdevices/c:/windows/winsxs/manifests/x86_microsoft.vc90.mfcloc_1fc8b3b9a1e18e3b_9.0.30729.6161_none_deadbeef.manifest
Executing rm -rf /home/txtsd/1TB/Games/kingdoms-of-amalur-reckoning/dosdevices/c:/windows/winsxs/manifests/x86_microsoft.vc90.openmp_1fc8b3b9a1e18e3b_9.0.30729.6161_none_deadbeef.manifest
Executing rm -rf /home/txtsd/1TB/Games/kingdoms-of-amalur-reckoning/dosdevices/c:/windows/winsxs/manifests/amd64_microsoft.vc90.atl_1fc8b3b9a1e18e3b_9.0.30729.6161_none_deadbeef.manifest
Executing rm -rf /home/txtsd/1TB/Games/kingdoms-of-amalur-reckoning/dosdevices/c:/windows/winsxs/manifests/amd64_microsoft.vc90.crt_1fc8b3b9a1e18e3b_9.0.30729.6161_none_deadbeef.manifest
Executing rm -rf /home/txtsd/1TB/Games/kingdoms-of-amalur-reckoning/dosdevices/c:/windows/winsxs/manifests/x86_microsoft.vc90.atl_1fc8b3b9a1e18e3b_9.0.30729.6161_none_deadbeef.manifest
Executing rm -rf /home/txtsd/1TB/Games/kingdoms-of-amalur-reckoning/dosdevices/c:/windows/winsxs/manifests/x86_microsoft.vc90.crt_1fc8b3b9a1e18e3b_9.0.30729.6161_none_deadbeef.manifest
Executing rm -rf /home/txtsd/1TB/Games/kingdoms-of-amalur-reckoning/dosdevices/c:/windows/winsxs/manifests/x86_microsoft.vc90.mfc_1fc8b3b9a1e18e3b_9.0.30729.6161_none_deadbeef.manifest
Executing rm -rf /home/txtsd/1TB/Games/kingdoms-of-amalur-reckoning/dosdevices/c:/windows/winsxs/manifests/x86_microsoft.vc90.mfcloc_1fc8b3b9a1e18e3b_9.0.30729.6161_none_deadbeef.manifest
Executing rm -rf /home/txtsd/1TB/Games/kingdoms-of-amalur-reckoning/dosdevices/c:/windows/winsxs/manifests/x86_microsoft.vc90.openmp_1fc8b3b9a1e18e3b_9.0.30729.6161_none_deadbeef.manifest
Executing rm -rf /home/txtsd/1TB/Games/kingdoms-of-amalur-reckoning/dosdevices/c:/windows/winsxs/manifests/amd64_microsoft.vc90.atl_1fc8b3b9a1e18e3b_9.0.30729.6161_none_deadbeef.manifest
Executing rm -rf /home/txtsd/1TB/Games/kingdoms-of-amalur-reckoning/dosdevices/c:/windows/winsxs/manifests/amd64_microsoft.vc90.crt_1fc8b3b9a1e18e3b_9.0.30729.6161_none_deadbeef.manifest
Executing rm -rf /home/txtsd/1TB/Games/kingdoms-of-amalur-reckoning/dosdevices/c:/windows/winsxs/manifests/x86_microsoft.vc90.atl_1fc8b3b9a1e18e3b_9.0.30729.6161_none_deadbeef.manifest
Executing rm -rf /home/txtsd/1TB/Games/kingdoms-of-amalur-reckoning/dosdevices/c:/windows/winsxs/manifests/x86_microsoft.vc90.crt_1fc8b3b9a1e18e3b_9.0.30729.6161_none_deadbeef.manifest
Executing rm -rf /home/txtsd/1TB/Games/kingdoms-of-amalur-reckoning/dosdevices/c:/windows/winsxs/manifests/x86_microsoft.vc90.mfc_1fc8b3b9a1e18e3b_9.0.30729.6161_none_deadbeef.manifest
Executing rm -rf /home/txtsd/1TB/Games/kingdoms-of-amalur-reckoning/dosdevices/c:/windows/winsxs/manifests/x86_microsoft.vc90.mfcloc_1fc8b3b9a1e18e3b_9.0.30729.6161_none_deadbeef.manifest
Executing rm -rf /home/txtsd/1TB/Games/kingdoms-of-amalur-reckoning/dosdevices/c:/windows/winsxs/manifests/x86_microsoft.vc90.openmp_1fc8b3b9a1e18e3b_9.0.30729.6161_none_deadbeef.manifest
Executing wine regedit C:\windows\Temp\override-dll.reg
Executing wine64 regedit C:\windows\Temp\override-dll.reg
Executing cd /home/txtsd/.cache/winetricks/vcrun2008
Executing wine vcredist_x86.exe
011c:fixme:clusapi:GetNodeClusterState ((null),0031ECC4) stub!

```
This is the point where it stalls and generates the folders. If I'm not running `--unattended` there's a window with `Verifying file: vcredist_x86` that's always at 0%.

Here's an `ls -alh` output of my external drive:
https://0x0.st/--fr.log
There are only 2 original folders in that dir named `Actual Folder {1,2}`

I'm not entirely sure this is winetricks' fault, so I'd like help with pinpointing the actual cause. As in, if another dependency might be causing it. I've deleted the folders and run winetricks several times to make sure winetricks is what was triggering/populating it.

**Comment by txtsd** [comment]
The stalled process finally fails with a `Cannot create` popup, and then a `Note: command wine vcredist_x86.exe returned status 82. Aborting.` popup.

**Comment by txtsd** [comment]
Just ran a couple of `WINEPREFIX=$prefixdir winetricks --unattended dotnet48 vcrun2019` and it doesn't trash the drive with folders, but it does create a temporary folder with the extracted files in it. It sometimes fails to delete this folder and leaves an empty one.

**Comment by austin987** [comment]
Thanks for the report; I had noticed this a couple months ago and hadn't looked into it until recently.

AFAICT, this is mostly due to a couple braindead installers (though obviously more may exist): dotnet35sp1 and gfw[1]. gfw isn't very commonly used, so this is more likely from dotnet35sp1. That said, you pointed out that dotnet48 also can do it, which I don't doubt, but I haven't personally seen.

In any event, I've made two fixes [2,3] that, for me, avoid the problem for most cases. I say avoid, not eliminate, because with the current design, there's no central way to avoid this problem (there's no distinction between running an arbitrary executable and an installer, which at least in my initial thoughts, would be the easiest way to avoid this). I've cleared my cache and run `make test`/`make check` without arbitrary directories left over.

1: Based on what `tests/winetricks-test` actually checks, i.e.,dlls/settings, not the complete collection of various apps/games.
2: https://github.com/Winetricks/winetricks/commit/7f7f3fd8298f737dcf7448819f2f0d5afda1501f
3: https://github.com/Winetricks/winetricks/commit/8d5194b3d879d0d31c7e33a6576aa86773658c83

TL;DR: this is currently fixed AFAIK, but could come up again (and those would be valid bugs). The generic problem isn't easily fixed without a redesign.

**Comment by txtsd** [comment]
Thanks. I won't be able to test it until the next release, after which I'll test it for most of my use cases again.

**Comment by austin987** [comment]
Sure, no worries. I agree that it's a problem, and an open to fixing it as it comes up, for sure.

I'm also open to a broader fix, but don't want to commit to doing it myself, at least until I find myself with more free time ;).
